compiler/hieFile/HieAst.hs

```diff
@@ -1280,7 +1280,7 @@ instance ToHie (Located OverlapMode) where
   toHie (L span _) = pure $ locOnly span
 
 instance ToHie a => ToHie (HsScaled GhcRn a) where
-  toHie = toHie . hsThing
+  toHie (HsScaled w t) = concatM [toHie (arrowToHsType w), toHie t]
```

Comment:
I don't know the intent of HIE files all that much. But is it ok to lose information this way? Should we be able to know whether to print back `a ->. b` rather than `a # 'One -> b`?

Reply:
I think yes - the constructors of `HieType` mirror `Type` and not `HsType`.